### PR TITLE
For #12240: Enable btime youtube-playback Fenix tests

### PR DESCRIPTION
**Pull request by alexandru-io**
Added the youtube-playback tests [from mozilla-central](https://searchfox.org/mozilla-central/search?q=raptor-youtube&path=testing%2Fraptor%2Fraptor%2Ftests%2Fbenchmark&case=false&regexp=false)

### CODE REVIEW

File: taskcluster/ci/browsertime/kind.yml [hunk 1]
```diff
@@ -232,3 +232,38 @@ jobs:
         test-name: google-search-restaurants
         treeherder:
             symbol: 'Btime(tp6m-28-c)'
+
+    youtube-playback:
+        test-name: browsertime-youtube-playback
+        treeherder:
+            symbol: 'Btime(ytp)'
+
+    youtube-playback-av1-sfr:
```
Comment (gmierz) [inline taskcluster/ci/browsertime/kind.yml:241]: Remove all of the 2020 variants for now, we'll add those later when they've been verified.

File: taskcluster/ci/browsertime/kind.yml [hunk 2]
```diff
@@ -232,3 +232,38 @@ jobs:
         test-name: google-search-restaurants
         treeherder:
             symbol: 'Btime(tp6m-28-c)'
+
+    youtube-playback:
+        test-name: browsertime-youtube-playback
+        treeherder:
+            symbol: 'Btime(ytp)'
```
Comment (gmierz) [inline taskcluster/ci/browsertime/kind.yml:239]: You need to add the args from here to this task: https://github.com/mozilla-mobile/fenix/blob/master/taskcluster/ci/raptor/kind.yml#L93-L98

File: taskcluster/ci/browsertime/kind.yml [hunk 3]
```diff
@@ -232,3 +232,38 @@ jobs:
         test-name: google-search-restaurants
         treeherder:
             symbol: 'Btime(tp6m-28-c)'
+
+    youtube-playback:
+        test-name: browsertime-youtube-playback
```
Comment (gmierz) [inline taskcluster/ci/browsertime/kind.yml:237]: Oh, one more thing: this test should be `youtube-playback`

File: taskcluster/ci/browsertime/kind.yml (hunk 1 above)
Comment (alexandru-io) [reply taskcluster/ci/browsertime/kind.yml:241]: 2020 variants meaning all except `youtube-playback`?

Comment (gmierz) [reply taskcluster/ci/browsertime/kind.yml]: yup that's right

**Comment by gmierz** [comment]
This looks good now, thanks! Running a test of it here: https://github.com/mozilla-mobile/fenix/pull/12344

**Comment by gmierz** [comment]
@alexandru-io you'll have to add `youtube-playback` to this list - odd that it was passing on try without this: https://searchfox.org/mozilla-central/source/testing/raptor/raptor/manifest.py#51

**Comment by gmierz** [comment]
@alexandru-io the tests failed and I think it's because of the `--cold` in this command: https://github.com/mozilla-mobile/fenix/blob/master/taskcluster/ci/browsertime/kind.yml#L84

If you re-define the `run:` field in the youtube task (but without the `--cold` in the command) then that should solve the issue.

File: taskcluster/ci/browsertime/kind.yml [hunk 4]
```diff
@@ -232,3 +232,31 @@ jobs:
         test-name: google-search-restaurants
         treeherder:
             symbol: 'Btime(tp6m-28-c)'
+
+    youtube-playback:
+        test-name: youtube-playback
+        treeherder:
```
Comment (gmierz) [inline taskcluster/ci/browsertime/kind.yml:238]: You'll need to add `run-visual-metrics: False` here.

**Comment by gmierz** [comment]
Can you add `--browsertime-no-ffwindowrecorder` to here as well: https://github.com/mozilla-mobile/fenix/blob/master/taskcluster/fenix_taskgraph/transforms/browsertime.py#L96

File: taskcluster/ci/browsertime/kind.yml [hunk 5]
```diff
@@ -232,3 +232,32 @@ jobs:
...
+            by-abi:
+                # Bug 1558456 - Stop tracking youtube-playback-test on motoG5 for >1080p cases
+                armeabi-v7a:
+                    - '--test-url-params=exclude=1,2,9,10,17,18,21,22,26,28,30,32,39,40,47,48,55,56,63,64,71,72,79,80,83,84,89,90,95,96'
+                default: []
+        run:
+            using: run-task
+            checkout: false
+            run-as-root: true
+                command:
```
Comment (gmierz) [inline taskcluster/ci/browsertime/kind.yml:251]: You'll have to unindent the command field and list.

**Comment by gmierz** [comment]
Great work @alexandru-io, tests are passing now! :D 

@JohanLorenzo if this looks good to you, would you be able to merge this please?

**Comment by gmierz** [comment]
Oh and the tests are here: https://github.com/mozilla-mobile/fenix/pull/12546

**Comment by JohanLorenzo** [comment]
I cannot land this PR because @alexandru-io isn't a collaborator on the fenix repo. Tests were done in https://github.com/mozilla-mobile/fenix/pull/12344. @liuche, could you bypass the Github check by merging the PR directly? I (luckily) don't have enough rights to do so.

**Comment by liuche** [comment]
Re-opening as #12760 to run CI!

**Pull request closed without merging**